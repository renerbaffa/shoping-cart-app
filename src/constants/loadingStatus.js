export const FAILED = 'FAILED';
export const RETRIEVED = 'RETRIEVED';
export const RETRIEVING = 'RETRIEVING';
