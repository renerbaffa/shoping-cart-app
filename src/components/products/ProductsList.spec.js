import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { GRID, LIST } from '../../constants/ViewOptions';

import ConnectedProductsList,
{ mapStateToProps, ProductsList } from './ProductsList';

import PROJECTS from '../../mocks/Products';
import convertToIdsAndContent from '../../normalizers/productsNormalize';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const NORMALIZED_PRODUCTS = convertToIdsAndContent(PROJECTS);
const FILTER = 'Fant';

jest.mock('./ProductCard', () => 'ProductCard');
jest.mock('./ProductItem', () => 'ProductItem');

describe('<ProductsList />', () => {
  let productsList;

  it('should render correctly on GRID view', () => {
    productsList = renderer.create(
      <ProductsList
        currentView={GRID}
        productIds={[1, 2]}
      />
    ).toJSON();
    expect(productsList).toMatchSnapshot();
  });

  it('should render correctly on LIST view', () => {
    productsList = renderer.create(
      <ProductsList
        currentView={LIST}
        productIds={[1, 2]}
      />
    ).toJSON();
    expect(productsList).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return filtered product ids', () => {
      const filteredProject =
        mapStateToProps(
          { products: NORMALIZED_PRODUCTS },
          { filterText: FILTER }
        ).productIds;

      expect(filteredProject).toHaveLength(1);
      expect(filteredProject[0]).toBe(24);
    });
  });

  it('should inject filter product id list', () => {
    const ids = [24];
    const store = mockStore({
      products: NORMALIZED_PRODUCTS,
    });

    const wrapper = shallow(
      <ConnectedProductsList
        store={store}
        filterText={FILTER}
        currentView={GRID}
      />
    );

    expect(wrapper.props().productIds).toEqual(ids);
  });
});
