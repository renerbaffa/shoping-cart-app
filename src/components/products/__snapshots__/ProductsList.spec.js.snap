// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsList /> should render correctly on GRID view 1`] = `
<div>
  <div
    className="ProductsList-container"
  >
    <div
      className="ProductCard-container"
      productId={1}
    >
      <img
        alt=""
        className="ProductCard-image"
        src=""
      />
      <div
        className="ProductCard-name"
      >
        
      </div>
      <div
        className="ProductCard-description"
      >
        
      </div>
      <button
        className="ProductCard-button Button"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="ProductCard-container"
      productId={2}
    >
      <img
        alt=""
        className="ProductCard-image"
        src=""
      />
      <div
        className="ProductCard-name"
      >
        
      </div>
      <div
        className="ProductCard-description"
      >
        
      </div>
      <button
        className="ProductCard-button Button"
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
</div>
`;

exports[`<ProductsList /> should render correctly on LIST view 1`] = `
<div>
  <div
    className="ProductsList-container"
  >
    <div>
      ProductItem
    </div>
    <div>
      ProductItem
    </div>
  </div>
</div>
`;
